XamarinCommunityToolkit/Effects/Touch/PlatformTouchEffect.android.cs

```diff
@@ -0,0 +1,340 @@
+					effect?.HandleHover(HoverStatus.Exited);
+					break;
+			}
+		}
+
+		void OnClick(object sender, EventArgs args)
+		{
+			if (effect?.IsDisabled ?? true)
+                return;
+
+			if (!IsAccessibilityMode)
+				return;
+
+			IsCanceled = false;
+			HandleEnd(TouchStatus.Completed);
+		}
+
+		void HandleEnd(TouchStatus status)
+		{
+			if (IsCanceled)
+				return;
+
+			IsCanceled = true;
+			if (effect.DisallowTouchThreshold > 0)
+				Group.Parent?.RequestDisallowInterceptTouchEvent(false);
+
+			effect?.HandleTouch(status);
+			effect?.HandleUserInteraction(TouchInteractionStatus.Completed);
+			EndRipple();
+		}
+
+		void StartRipple(float x, float y)
+		{
+			if (effect?.IsDisabled ?? true)
+                return;
+
+			if (effect.CanExecute && effect.NativeAnimation && viewOverlay?.Background is RippleDrawable)
+			{
+				UpdateRipple();
+				viewOverlay.BringToFront();
+				ripple.SetHotspot(x, y);
+				viewOverlay.Pressed = true;
+			}
+		}
+
+		void EndRipple()
+		{
+			if (effect?.IsDisabled ?? true)
+                return;
+
+			if (viewOverlay?.Pressed ?? false)
+				viewOverlay.Pressed = false;
+		}
+
+		void CreateRipple()
+		{
+			var background = View?.Background;
+
+			if (background is RippleDrawable)
+			{
+				ripple = (RippleDrawable)background.GetConstantState().NewDrawable();
+				return;
+			}
+
+			var noBackground = Element is Layout || background == null;
+			ripple = new RippleDrawable(GetColorStateList(), noBackground ? null : background, noBackground ? new ColorDrawable(Color.White) : null);
+			UpdateRipple();
+		}
+
+		void UpdateRipple()
+		{
+			if (effect?.IsDisabled ?? true)
+				return;
+
+			if (effect.NativeAnimationColor == rippleColor && effect.NativeAnimationRadius == rippleRadius)
+				return;
+
+			rippleColor = effect.NativeAnimationColor;
+			rippleRadius = effect.NativeAnimationRadius;
+			ripple.SetColor(GetColorStateList());
+			if (AndroidOS.Build.VERSION.SdkInt >= AndroidOS.BuildVersionCodes.M)
+				ripple.Radius = (int)(View.Context.Resources.DisplayMetrics.Density * effect.NativeAnimationRadius);
+		}
+
+		ColorStateList GetColorStateList()
+		{
+			int color;
+			var defaultcolor = effect.NativeAnimationColor;
+			if (defaultcolor != Forms.Color.Default)
+				color = defaultcolor.ToAndroid();
+			else
+				color = Color.Argb(64, 128, 128, 128);
+
+			return new ColorStateList(
+				new[] { new int[] { } },
+				new[] { color, });
+		}
+
+		void LayoutChange(object sender, AView.LayoutChangeEventArgs e)
+		{
+			var group = (ViewGroup)sender;
+			if (group == null || (Group as IVisualElementRenderer)?.Element == null)
+                return;
+
+			viewOverlay.Right = group.Width;
+			viewOverlay.Bottom = group.Height;
+		}
+
+		sealed class AccessibilityListener : Java.Lang.Object,
+											 AccessibilityManager.IAccessibilityStateChangeListener,
+											 AccessibilityManager.ITouchExplorationStateChangeListener
+		{
+			readonly PlatformTouchEffect platformTouchEffect;
```

Comment:
Since here we are in the Android world, I think that we can make this a `WeakReference`. In my head, this can avoid memory leaks.

Reply:
Worked around via Dispose method (clear this ref there. Made it non-read-only)